.github/workflows/deploy_dbt_docs.yaml

```diff
@@ -0,0 +1,71 @@
+name: deploy-dbt-docs
+
+on:
+  workflow_run:
+    workflows: [build-and-test-dbt]
+    branches: [master, data-catalog]
+    types: [completed]
```

Comment:
The idea here is that we only want this workflow to run after the `build-and-test-dbt` workflow has completed on the `master` or `data-catalog` branch. Unfortunately, there's no `type` that corresponds to a _successful_ build, so the best we can do is check for the `completed` type and gate each job step behind the `github.event.workflow_run.conclusion == 'success'` expression (see [this SO thread](https://stackoverflow.com/a/64733705) and the [GitHub Actions feature request](https://github.com/actions/runner/issues/662) that it links to).

Reply:
**suggestion (blocking):** I believe the entire `deploy-dbt-docs` job can be gated with a single if statement. See example here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow. I would refactor for simplicity if possible, but I might be missing some caveat here.